code/modules/shuttle/supply.dm

```diff
@@ -223,8 +226,44 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
 		msg += export_text + "\n"
 		D.adjust_money(ex.total_value[E])
 
+	SSeconomy.export_total += (D.account_balance - presale_points)
 	SSshuttle.centcom_message = msg
 	investigate_log("Shuttle contents sold for [D.account_balance - presale_points] credits. Contents: [ex.exported_atoms ? ex.exported_atoms.Join(",") + "." : "none."] Message: [SSshuttle.centcom_message || "none."]", INVESTIGATE_CARGO)
 
+/// Generates a box of mail depending on our exports and imports.
+/obj/docking_port/mobile/supply/proc/mail()
+	if(SSeconomy.mail_waiting <= 0)
+		return FALSE
+
+	var/mail_recipients = list()
+	for(var/mob/living/carbon/human/H in shuffle(GLOB.alive_mob_list))
+		if(!H.client || H.stat == DEAD)
+			continue
+		mail_recipients += list(H)
```

Comment:
This should use a GLOB.living_humans or such not this

Reply:
That glob doesn't seem to exist. It's borrowed code from spontaneous appendicitis / the ghost orbit UI. There's about 300 mobs in this GLOB on a 50 player server.

Reply:
the bigger worry is the use of shuffle() as it's not really necesary. The list used doesn't matter too much but yeah using player_list is probably cheaper